lighthouse-core/lib/lh-error.js

```diff
@@ -96,6 +103,74 @@ class LighthouseError extends Error {
     const error = new Error(`Protocol error ${errMsg}`);
     return Object.assign(error, {protocolMethod: method, protocolError: protocolError.message});
   }
+
+  /**
+   * A JSON.stringify replacer to serialize LHErrors and (as a fallback) Errors.
+   * Returns a simplified version of the error object that can be reconstituted
+   * as a copy of the original error at parse time.
+   * @param {Error|LighthouseError} err
+   * @return {SerializedBaseError|SerializedLighthouseError}
+   */
+  static stringifyReplacer(err) {
+    if (err instanceof LighthouseError) {
+      // Remove class props so that remaining values were what was passed in as `properties`.
+      // eslint-disable-next-line no-unused-vars
+      const {name, code, message, friendlyMessage, lhrRuntimeError, stack, ...properties} = err;
+
+      return {
+        sentinel: LHERROR_SENTINEL,
+        code,
+        stack,
+        ...properties,
+      };
+    }
+
+    // We still have some errors that haven't moved to be LHErrors, so serialize them as well.
```

Comment:
any error that happens unexpectedly will always be a non-LHError, so it's a condition that's likely to exist forever :)

```suggestion
    // Unexpected errors won't be `LHError`, but we still want to serialize them as well.
```

Reply:
we could leave the line in if we want to also nudge folks to look into the still expected ones that aren't LHError yet?

Reply:
> we could leave the line in if we want to also nudge folks to look into the still expected ones that aren't LHError yet?

no, I think you're right. We should have some nudges but probably not here :)